stomp_moveit/src/cost_functions/collision_check.cpp

```diff
@@ -238,21 +229,22 @@ bool CollisionCheck::computeCosts(const Eigen::MatrixXd& parameters,
   {
     if(!skip_next_check)
     {
-      robot_state_->setJointGroupPositions(joint_group,parameters.col(t));
-      robot_state_->update();
+      robot_state::RobotState robot_state(*robot_state_);
```

Comment:
The `computeCosts` function is invoked many many times during a single iteration and so instantiating objects within it could render it more inefficient.

Reply:
This was required for parallel implementation of the code. It cannot change local members from multiple threads.